File: nfcservice/src/main/java/br/com/claro/hackaton/nfcservice/NfcApiModule.java
package br.com.claro.hackaton.nfcservice;

import android.support.annotation.Nullable;

import com.google.gson.FieldNamingPolicy;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by Leandro on 23/01/2018.
 */

public class NfcApiModule {

    private static final String BASE_URL = "https://api.themoviedb.org/3/";
    private static Retrofit retrofit;

    public static void setRetrofit(@Nullable LoggingInterceptor.LogLevel logLevel) {

        LoggingInterceptor interceptor = new LoggingInterceptor(logLevel);
        OkHttpClient.Builder builder = new OkHttpClient.Builder();
        builder.addInterceptor(interceptor);
        builder.addInterceptor(chain -> chain.proceed(chain.request()));
//        builder.addInterceptor(chain -> {
//            Request request = chain.request().newBuilder()
//                    .addHeader("Authorization", String.format("Bearer %s", API_KEY)).build();
//            return chain.proceed(request);
//        });
        OkHttpClient okClient = builder.build();
        retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(okClient)
                .addConverterFactory(GsonConverterFactory.create(getDefaultGsonBuilder()))
                .build();
    }

    public static Gson getDefaultGsonBuilder() {
        return new GsonBuilder()
                .setFieldNamingPolicy(FieldNamingPolicy.IDENTITY)
                .create();
    }

    public static Retrofit getRetrofit() {
        return retrofit;
    }

}

File: nfcservice/src/main/java/br/com/claro/hackaton/nfcservice/NfcApiModule.kt
package br.com.claro.hackaton.nfcservice

import com.google.gson.FieldNamingPolicy
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import okhttp3.OkHttpClient
import org.threeten.bp.Clock
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by Leandro on 22/02/2018.
 */
internal const val BASE_URL: String = "https://api.github.com/"

/**
 * Created by gibran.lyra on 23/08/2017.
 */
object GitHubApiModule {
    lateinit var retrofit: Retrofit private set

    fun setRetrofit(logLevel: LoggingInterceptor.Level = LoggingInterceptor.Level.FULL) {
        val builder = OkHttpClient.Builder()
        val loggingInterceptor = LoggingInterceptor(Clock.systemDefaultZone(), logLevel)
        builder.addInterceptor(loggingInterceptor)

        val okClient = builder.build()
        retrofit = Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(okClient)
                .addConverterFactory(GsonConverterFactory.create(getGsonBuilder()))
//                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build()
    }

    fun getGsonBuilder(): Gson {
        return GsonBuilder()
                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
                .setPrettyPrinting()
                .create()
    }
}